wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java

```diff
@@ -91,32 +96,20 @@ public User create(User newUser, boolean isTemporary) throws ConflictException,
                                            newUser.getName(),
                                            firstNonNull(newUser.getPassword(), generate("", PASSWORD_LENGTH)),
                                            newUser.getAliases());
-        try {
-            userDao.create(user);
-            profileDao.create(new ProfileImpl(user.getId()));
-            preferencesDao.setPreferences(user.getId(), ImmutableMap.of("temporary", Boolean.toString(isTemporary),
-                                                                        "codenvy:created", Long.toString(currentTimeMillis())));
-        } catch (ConflictException | ServerException x) {
-            // optimistic rollback(won't remove profile if userDao.remove failed)
-            // remove operation is not-found-safe so if any exception
-            // during the either user or profile creation occurs remove all entities
-            // NOTE: this logic must be replaced with transaction management
-            try {
-                userDao.remove(user.getId());
-                profileDao.remove(user.getId());
-                preferencesDao.remove(user.getId());
-            } catch (ServerException rollbackEx) {
-                LOG.error(format("An attempt to clean up resources due to user creation failure was unsuccessful." +
-                                 "Now the system may be in inconsistent state. " +
-                                 "User with id '%s' must not exist",
-                                 user.getId()),
-                          rollbackEx);
-            }
-            throw x;
-        }
+        doCreate(user, isTemporary);
+        eventService.publish(new UserCreatedEvent(user));
         return user;
     }
 
+    @Transactional(rollbackOn = {RuntimeException.class, ApiException.class})
+    protected void doCreate(UserImpl user, boolean isTemporary) throws ConflictException, ServerException {
+        userDao.create(user);
+        eventService.publish(new PostUserPersistedEvent(new UserImpl(user))).propagateException();
```

Comment:
Mb It'd better to publish event after profile creation and preferences setting. I'd allow listener to put something into preferences or profile attributes, otherwise we'll need different events for profile and preferences when there will be such listener.
For me it's OK to listen user persisted event and put something into profile because UserManager#create -> create profile automatically and UserManager publish this event. So it's the same as expect that profile will exist after user creation by UserManager#create
WDYT?